pkg/controller/nodemaintenance/nodemaintenance_controller.go

```diff
@@ -245,9 +314,137 @@ func (r *ReconcileNodeMaintenance) stopNodeMaintenance(nodeName string) error {
 		return err
 	}
 
+	if r.isLeaseSupported {
+		log.Info("Lease object supported, deleting lease")
+		if err := r.deleteLease(nodeName); err != nil {
+			return err
+		}
+	}
+
 	return nil
 }
 
+func (r *ReconcileNodeMaintenance) deleteLease(nodeName string) error {
+
+		nName := apitypes.NamespacedName{Namespace: LeaseNamespace, Name: nodeName}
+		lease := &coordv1beta1.Lease{}
+
+		if err := r.client.Get(context.TODO(), nName, lease); err != nil {
+			return err
+		}
+		r.client.Delete(context.TODO(), lease);
+		return nil
+}
+
+func (r *ReconcileNodeMaintenance) obtainLease(node *corev1.Node, durationInSeconds int) (bool, reconcile.Result) {
+
+	lease, needUpdate, err := r.createLease(node, durationInSeconds)
+
+	if err != nil {
+		log.Errorf("MaintOn: failed to create lease error=%v", err)
+		return false, reconcile.Result{Requeue: true, RequeueAfter: RequeWaitTime}
+	}
+
+	if needUpdate == false {
+		return true, reconcile.Result{}
+	}
+
+	log.Info("MaintOn: update lease - lease duration change requested")
+
+	if _, err = r.updateLease(node, lease, durationInSeconds, TransitionNoChange, false); err != nil {
```

Comment:
If the lease exists prior to `createLease()` being called, then `lease` will be `nil` at this point and `updateLease()` will always fail.

Reply:
fixed, added a test for that situation.